libs/cookies/src/lib/services/cookie/cookie.service.ts

```diff
@@ -0,0 +1,200 @@
+import { isPlatformBrowser } from '@angular/common';
+import * as CookieConsent from 'vanilla-cookieconsent';
+
+import { BehaviorSubject, Observable, Subject, combineLatest, map, startWith } from 'rxjs';
+import {
+	NgxCookieCategories,
+	NgxCookieChangedEvent,
+	NgxCookieConfiguration,
+	NgxCookieEvent,
+	NgxCookieLanguageConfiguration,
+} from '../../types';
+
+/**
+ * A service that implements the CookieConsent from https://cookieconsent.orestbida.com and provides the necessary Observables to handle this setup.
+ */
+@Injectable({ providedIn: 'root' })
+export class NgxCookieService {
+	/**
+	 * Subject to hold the onFirstConsent event
+	 */
+	private readonly firstCookiesConsentedSubject: Subject<NgxCookieEvent> = new Subject();
+
+	/**
+	 * Subject to hold the onConsent event
+	 */
+	private readonly cookiesConsentedSubject: Subject<NgxCookieEvent> = new Subject();
+
+	/**
+	 * Subject to hold the consentChanged event
+	 */
+	private readonly cookiesConsentChangedSubject: Subject<NgxCookieChangedEvent> = new Subject();
+
+	/**
+	 * Subject to hold the modalVisible event
+	 */
+	private readonly modalVisibleSubject: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(
+		false
+	);
+
+	/**
+	 * An event triggered only the very first time that the user expresses their choice of consent
+	 */
+	public readonly firstCookiesConsented$: Observable<NgxCookieEvent> =
+		this.firstCookiesConsentedSubject.asObservable();
+
+	/**
+	 * An event triggering the choice of consent on every page load.
+	 */
+	public readonly cookiesConsented$: Observable<NgxCookieEvent> =
+		this.cookiesConsentedSubject.asObservable();
+
+	/**
+	 * An event triggered when the user modifies their preferences and only if consent has already been provided.
+	 */
+	public readonly cookiesConsentChanged$: Observable<NgxCookieChangedEvent> =
+		this.cookiesConsentChangedSubject.asObservable();
+
+	/**
+	 * Whether the cookies modal is currently visible
+	 */
+	public readonly modalVisible$: Observable<boolean> = this.modalVisibleSubject.asObservable();
+
+	constructor(@Inject(PLATFORM_ID) private platformId: string) {}
+
+	/**
+	 * Sets up the CookieConsent.
+	 *
+	 * For more information of on this configuration, check the documentation
+	 * https://cookieconsent.orestbida.com/essential/getting-started.html
+	 *
+	 * @param categories - The categories of cookies
+	 * @param language - The language settings of the cookies
+	 * @param configuration - An optional set configuration settings
+	 */
+	setupCookiesHandler(
+		categories: NgxCookieCategories,
+		language: NgxCookieLanguageConfiguration,
+		configuration?: NgxCookieConfiguration
+	): void {
+		// Iben: If we're not in the browser, we early exit, so server-side rendering can be enabled
+		if (!isPlatformBrowser(this.platformId)) {
+			return;
+		}
+
+		// Iben: If no categories were provided, we early exit and throw an error.
+		if (!categories || Object.keys(categories).length === 0) {
+			console.error(
```

Comment:
Maakt dit iets kapot? Anders zou ik daar een warning van durven maken.

Reply:
It does indeed, if you pass no categories the no popup gets shown, hence why.

Reply:
Ok that's fair. Thanks for the info.

Reply:
I updated the comment however, so that it's clearer to us why it throws an error 👍